pkg/backend/display/progress.go

```diff
@@ -856,8 +856,16 @@ func (display *ProgressDisplay) processNormalEvent(event engine.Event) {
 		// Note: we should probably make sure we don't get any prelude events
 		// once we start hearing about actual resource events.
 
-		payload := event.Payload.(engine.PreludeEventPayload)
-		display.writeSimpleMessage(renderPreludeEvent(payload, display.opts))
+		if !display.isTerminal {
+			// If we are in a terminal, then we may have already rendered progress events, and so we
```

Comment:
another option is to forward this as an ephemeral message to the root resource.  so it still shows up somewhere, but doesn't impact the display.  

But this is a reasonable place to start.

Reply:
> another option is to forward this as an ephemeral message to the root resource. so it still shows up somewhere, but doesn't impact the display.

Yeah, this is what I'd expect us to do

Reply:
if this contains important data, we could also treat this like we do summary-data, and put it at the end.  It's unclear what information in this is important though.

Reply:
I'm honestly not sure anyone knows that `--show-config` exists - and without it set this does nothing (except introduce the bug we are fixing here).

But when a user explicitly passes `--show-config` we render something like this today:

```
Previewing update (dev):
     Type                 Name      Plan     Info
     pulumi:pulumi:Stack  perf-dev           warning: resource plugin aws is expected to have version >=1.20.0, but has 1.19.0-alpha.1579202679+gd1b5c8a47; the wrong versi
Configuration:
    aws:region: us-east-1

     Type                 Name      Plan     Info
     pulumi:pulumi:Stack  perf-dev           2 warnings
 
Diagnostics:
  pulumi:pulumi:Stack (perf-dev):
    warning: resource plugin aws is expected to have version >=1.20.0, but has 1.19.0-alpha.1579202679+gd1b5c8a47; the wrong version may be on your path, or this may be a bug in the plugin
    warning: resource plugin aws is expected to have version >=1.20.0, but has 1.19.0-alpha.1579202679+gd1b5c8a47; the wrong version may be on your path, or this may be a bug in the plugin
```

These two lines are the key piece of that:

```
Configuration:
    aws:region: us-east-1
```

We *could* show that at the end - though I think it is really more of an input thing which is intended to be shown at the beginning.  It's just that we don't get the data early enough to actually show it before beginning to render the table.

Reply:
Going to update so that it prints this 

```
$ /opt/pulumi/bin/pulumi up --show-config
Previewing update (dev):
     Type                 Name      Plan     Info
     pulumi:pulumi:Stack  perf-dev           1 message
 
Diagnostics:
  pulumi:pulumi:Stack (perf-dev):
    Configuration:
        aws:region: us-east-1
```

Reply:
👍 Def prefer this way.